test/e2e/framework/console-operator.go

```diff
@@ -0,0 +1,390 @@
+	return cr.Spec.ManagementState == operatorsv1.Unmanaged
+}
+
+func isOperatorRemoved(cr *operatorsv1.Console) bool {
+	return cr.Spec.ManagementState == operatorsv1.Removed
+}
+
+type operatorStateReactionFn func(cr *operatorsv1.Console) bool
+
+func ensureConsoleIsInDesiredState(t *testing.T, client *ClientSet, state operatorsv1.ManagementState) error {
+	t.Helper()
+	var operatorConfig *operatorsv1.Console
+	// var checkFunc func()
+	var checkFunc operatorStateReactionFn
+
+	switch state {
+	case operatorsv1.Managed:
+		checkFunc = isOperatorManaged
+	case operatorsv1.Unmanaged:
+		checkFunc = isOperatorUnmanaged
+	case operatorsv1.Removed:
+		checkFunc = isOperatorRemoved
+	}
+
+	err := wait.Poll(1*time.Second, AsyncOperationTimeout, func() (stop bool, err error) {
+		operatorConfig, err = client.Operator.Consoles().Get(consoleapi.ConfigResourceName, metav1.GetOptions{})
+		if err != nil {
+			return false, err
+		}
+		return checkFunc(operatorConfig), nil
+	})
+	if err != nil {
+		DumpObject(t, "the latest observed state of the console resource", operatorConfig)
+		DumpOperatorLogs(t, client)
+		return fmt.Errorf("failed to wait to change console operator state to 'Removed': %s", err)
+	}
+	return nil
+}
+
+func ManageConsole(t *testing.T, client *ClientSet) error {
+	t.Helper()
+	operatorConfig, err := client.Operator.Consoles().Get(consoleapi.ConfigResourceName, metav1.GetOptions{})
+	if err != nil {
+		return err
+	}
+
+	if isOperatorManaged(operatorConfig) {
+		t.Logf("console operator already in 'Managed' state")
+		return nil
+	}
+
+	t.Logf("changing console operator state to 'Managed'...")
+
+	_, err = client.Operator.Consoles().Patch(consoleapi.ConfigResourceName, types.MergePatchType, []byte(`{"spec": {"managementState": "Managed"}}`))
+	if err != nil {
+		return err
+	}
+	if err := ensureConsoleIsInDesiredState(t, client, operatorsv1.Managed); err != nil {
+		return fmt.Errorf("unable to change console operator state to 'Managed': %s", err)
+	}
+
+	err = ConsoleResourcesAvailable(client)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	return nil
+}
+
+func UnmanageConsole(t *testing.T, client *ClientSet) error {
+	t.Helper()
+	operatorConfig, err := client.Operator.Consoles().Get(consoleapi.ConfigResourceName, metav1.GetOptions{})
+	if err != nil {
+		return err
+	}
+
+	if isOperatorUnmanaged(operatorConfig) {
+		t.Logf("console operator already in 'Unmanaged' state")
+		return nil
+	}
+
+	t.Logf("changing console operator state to 'Unmanaged'...")
+
+	_, err = client.Operator.Consoles().Patch(consoleapi.ConfigResourceName, types.MergePatchType, []byte(`{"spec": {"managementState": "Unmanaged"}}`))
+	if err != nil {
+		return err
+	}
+	if err := ensureConsoleIsInDesiredState(t, client, operatorsv1.Unmanaged); err != nil {
+		return fmt.Errorf("unable to change console operator state to 'Unmanaged': %s", err)
+	}
+
+	return nil
+}
+
+func RemoveConsole(t *testing.T, client *ClientSet) error {
```

Comment:
Not necessary to export, since it's only used in the `MustRemoveConsole()`

Reply:
+1